### PR TITLE
Parmest examples

### PR DESCRIPTION
**Pull request by kaklise**
## Summary/Motivation:
This PR adds basic tests for the parmest examples.  These are mainly smoke tests, but assert statements have been added when an expected solution is available.  As part of this update, the example files were cleaned up and documentation related to parmest was updated. The scipy example that computes covariance was moved to tests.

## Changes proposed in this PR:
- Cleaned up parmest examples
- Added tests for parmest examples

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: pyomo/contrib/parmest/examples/reactor_design/datarec_example.py [hunk 1]
```diff
@@ -15,70 +15,79 @@
...
+                  'k3': 1.0/6000.0}
+    print(theta_real)
 
-pest = parmest.Estimator(reactor_design_model, data_rec, theta_names, SSE)
-obj, theta = pest.theta_est()
-print(obj)
-print(theta)
-print(theta_real)
+if __name__ == "__main__":
+    main()
```
Comment (mrmundt) [inline pyomo/contrib/parmest/examples/reactor_design/datarec_example.py:93]: Newline needed

File: pyomo/contrib/parmest/examples/reactor_design/leaveNout_example.py [hunk 2]
```diff
@@ -10,68 +10,77 @@
...
+                              title= 'Alpha: '+ str(alpha) + ', '+ \
+                              str(theta_est_N.loc[0,alpha]))
+        
+    # Extract the percent of points that are within the alpha region
+    r = [results[i][1].loc[0,alpha] for i in range(lNo_samples)]
+    percent_true = sum(r)/len(r)
+    print(percent_true)
+
+if __name__ == "__main__":
+    main()
```
Comment (mrmundt) [inline pyomo/contrib/parmest/examples/reactor_design/leaveNout_example.py:86]: Newline needed

File: pyomo/contrib/parmest/examples/reactor_design/reactor_design.py [hunk 3]
```diff
@@ -74,4 +73,7 @@ def reactor_design_model(data):
     
     results = pd.DataFrame(results, columns=['sv', 'caf', 'ca', 'cb', 'cc', 'cd'])
     print(results)
+    
+if __name__ == "__main__":
+    main()
```
Comment (mrmundt) [inline pyomo/contrib/parmest/examples/reactor_design/reactor_design.py:78]: Newline needed.

File: pyomo/contrib/parmest/examples/reactor_design/timeseries_data_example.py [hunk 4]
```diff
@@ -9,34 +9,43 @@
...
+    obj, theta = pest.theta_est()
+    print(obj)
+    print(theta)
 
-pest = parmest.Estimator(reactor_design_model, data_ts, theta_names, SSE_timeseries)
-obj, theta = pest.theta_est()
-print(obj)
-print(theta)
+if __name__ == "__main__":
+    main()
```
Comment (mrmundt) [inline pyomo/contrib/parmest/examples/reactor_design/timeseries_data_example.py:51]: Newline needed.

File: pyomo/contrib/parmest/examples/semibatch/parameter_estimation_example.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    # for sum of squared error that will be used in parameter estimation
+            
+    pest = parmest.Estimator(generate_model, data, theta_names)
+    
+    obj, theta = pest.theta_est()
+    print(obj)
+    print(theta)
+
+if __name__ == '__main__':
+    main()
```
Comment (mrmundt) [inline pyomo/contrib/parmest/examples/semibatch/parameter_estimation_example.py:40]: Newline needed.

**Review by mrmundt — CHANGES_REQUESTED** [review]
This overall looks good - just some minor changes requested.

**Review by blnicho — APPROVED** [review]

**Review by mrmundt — APPROVED** [review]

**Pull request merged**